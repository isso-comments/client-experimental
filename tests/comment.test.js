const $ = require('lib/dom');
const config = require('config');
const i18n = require('i18n');
const offset = require('offset');
const template = require('template');

const commentHelper = require('comment');

const comment = {
  "id": 2,
  "created": 1651744800.0, // = 2022-05-05T10:00:00.000Z
  "mode": 1,
  "text": "<p>A comment with</p>\n<pre><code>code blocks\nNew line: preformatted\n\nDouble newline\n</code></pre>",
  "author": "John",
  "website": "http://website.org",
  "hash": "4505c1eeda98",
  "parent": null,
}

test('Rendered comment should match snapshot', () => {

  document.body.innerHTML =
    '<div id=isso-thread></div>' +
    '<script src="http://isso.api/js/embed.min.js" data-isso="/"></script>';
  issoThread = $('#isso-thread');
  issoThread.append('<div id="isso-root"></div>');


  let conf = {
    "avatar": true,
    "avatar-bg": "#f0f0f0",
    "avatar-fg": "#9abf88 #5698c4 #e279a3 #9163b6 #be5168 #f19670 #e4bf80 #447c69",
    "css": true,
    "css-url": null,
    "default-lang": "en",
    "feed": false,
    "gravatar": false,
    "lang": "",
    "max-comments-nested": 5,
    "max-comments-top": "inf",
    "page-author-hashes": "",
    "reply-notifications": false,
    "reply-notifications-default-enabled": false,
    "reply-to-self": false,
    "require-author": false,
    "require-email": false,
    "reveal-on-click": 5,
    "vote": true,
    "vote-levels": null
  };

  conf['langs'] = ['en-US', 'en', 'en', 'en'];

  let i18n_ = new i18n.I18n();
  i18n_.config = conf;
  i18n_.initTranslations();

  let template_ = new template.Template();
  template_.templateVars['conf'] = conf;
  template_.templateVars['i18n'] = i18n_;
  template_.templateVars['svg'] = {'arrow-up': '<svg></svg>', 'arrow-down': '<svg></svg>'};

  let fakeDate = new Date('2022-05-05T11:00:00.000Z'); // comment date + 1h
  offset.update(fakeDate);

  let _comment = new commentHelper.Comment(null, null, conf,
    i18n_, template_);
  _comment.insertComment(comment, false);

  expect($('.isso-comment').outerHTML).toMatchSnapshot();
});
