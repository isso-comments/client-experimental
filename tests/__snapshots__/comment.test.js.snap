// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered comment should match snapshot 1`] = `
"<div class=\\"isso-comment isso-no-votes\\" id=\\"isso-2\\" data-hash=\\"4505c1eeda98\\"><div class=\\"isso-avatar\\"><svg version=\\"1.1\\" viewBox=\\"0 0 48 48\\" preserveAspectRatio=\\"xMinYMin meet\\" shape-rendering=\\"crispEdges\\" data-hash=\\"4505c1eeda98\\"><rect x=\\"0\\" y=\\"0\\" width=\\"56\\" height=\\"56\\" style=\\"fill: #f0f0f0\\"></rect><rect x=\\"4\\" y=\\"4\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"36\\" y=\\"4\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"4\\" y=\\"20\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"36\\" y=\\"20\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"4\\" y=\\"28\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"36\\" y=\\"28\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"12\\" y=\\"4\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"28\\" y=\\"4\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"12\\" y=\\"12\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"28\\" y=\\"12\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"12\\" y=\\"28\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"28\\" y=\\"28\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"20\\" y=\\"4\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"20\\" y=\\"28\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect><rect x=\\"20\\" y=\\"36\\" width=\\"8\\" height=\\"8\\" style=\\"fill: #\\"></rect></svg></div><div class=\\"isso-text-wrapper\\"><div class=\\"isso-comment-header\\" role=\\"meta\\"><a class=\\"isso-author\\" href=\\"http://website.org\\" rel=\\"nofollow\\">John</a><span class=\\"isso-spacer\\">â€¢</span><a class=\\"isso-permalink\\" href=\\"#isso-2\\"><time title=\\"Thu May 05 2022 10:00:00 GMT+0000 (Coordinated Universal Time)\\" datetime=\\"2022-05-05T10:00:00Z\\">an hour ago</time></a><span class=\\"isso-note\\"></span></div><div class=\\"isso-text\\"><p>A comment with</p>
<pre><code>code blocks
New line: preformatted

Double newline
</code></pre></div><div class=\\"isso-comment-footer\\"><span class=\\"isso-votes\\"></span><a class=\\"isso-upvote\\" href=\\"#\\"><svg></svg></a><span class=\\"isso-spacer\\">|</span><a class=\\"isso-downvote\\" href=\\"#\\"><svg></svg></a><a class=\\"isso-reply\\" data-textreply=\\"Reply\\" data-textclose=\\"Close\\" href=\\"#\\">Reply</a></div></div><div class=\\"isso-follow-up\\"></div></div>"
`;
